cpp/src/plasma/malloc.cc

```diff
@@ -122,6 +122,13 @@ int create_buffer(int64_t size) {
       return -1;
     }
   }
+  int ret = dup(fd);
```

Comment:
Can you describe why this is needed?

Reply:
Sure. `file` needs to be closed, otherwise it is a memory leak. Closing it closes the descriptor `fd` but the descriptor is what this function is supposed to return. So we need a copy, close `file` and thus `fd`, and return the copy.
The real question is: why do we even need `file`? But there's a check based on that stream so I assume there is a good reason.

Reply:
IMHO we could remove the call to `fopen` and the related check entirely. The resulting `file` is never used for reading/writing. It seems that replacing `fopen` with `open`/`_open` will give us a duplicate descriptor, and thus we will still have to close one of them: the new one of `fd`. If that's OK, before I replace `fopen` with `open`, could you please tell me why is it needed there in the first place?